seal5/transform/detect_registers.py

```diff
@@ -42,13 +49,16 @@ def detect_registers(set_def: seal5_model.Seal5InstructionSet):
         if size is None:
             raise NotImplementedError(f"Register with unknown size: {name}")
         if size > 1:  # TODO: check this!
-            names = [f"{name}{i}" for i in range(size)]
-            group = seal5_model.Seal5RegisterGroup(names, size, width, reg_class)
+            names = [f"{name}{i}" for i in range(mem.data_range.lower, mem.data_range.upper + 1)]
+            # print("names", names)
+            group = seal5_model.Seal5RegisterGroup(names, size, width, signed, reg_class)
+            # print("group", group)
             set_def.register_groups[name] = group
             for reg in group.registers:
-                set_def.registers[name] = reg
+                set_def.registers[reg.name] = reg
         else:
-            reg = seal5_model.Seal5Register(name, size, width, reg_class)
+            reg = seal5_model.Seal5Register(name, size, width, signed, reg_class)
```

Comment:
(not so much of a review comment & probably I should just play with this code in the debugger..) - I didn't understand what about the custom register support drove the addition of signedness here, since the standard flags and GPR registers are also unsigned, not just the S4E_MAC::ACC ?

Reply:
The change was unrelated, I thought it would make sense… However Seal5 does not use it at all. I might remove it again to avoid confusion.

Reply:
It's not a bad change, I was just groping in the fog to understand how the custom register contrasted to the standard ones. So in itself I don't think it's confusing or should be removed